modules/core/src/nan_mask.simd.hpp

```diff
@@ -0,0 +1,435 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+
+
+#include "precomp.hpp"
+
+namespace cv {
+
+typedef void (*PatchNanFunc)(uchar* tptr, size_t len, double newVal);
+typedef void (*FiniteMaskFunc)(const uchar *src, uchar *dst, size_t total);
+
+CV_CPU_OPTIMIZATION_NAMESPACE_BEGIN
+
+PatchNanFunc getPatchNanFunc(bool isDouble);
+FiniteMaskFunc getFiniteMaskFunc(bool isDouble, int cn);
+
+#ifndef CV_CPU_OPTIMIZATION_DECLARATIONS_ONLY
+
+static void patchNaNs_32f(uchar* ptr, size_t len, double newVal)
+{
+    int32_t* tptr = (int32_t*)ptr;
+    Cv32suf val;
+    val.f = (float)newVal;
+
+    size_t j = 0;
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+    v_int32 v_pos_mask = vx_setall_s32(0x7fffffff), v_exp_mask = vx_setall_s32(0x7f800000);
+    v_int32 v_val = vx_setall_s32(val.i);
+
+    size_t cWidth = (size_t)VTraits<v_int32>::vlanes();
+    for (; j + cWidth*2 <= len; j += cWidth*2)
```

Comment:
`for` condition in SIMD loops should be `j < loop_invariant`.
Don't compute on the fly. See other SIMD loops.

Reply:
Fixed